[Term Entry] PyTorch function: tensor_split() (#5948)
Codecademy/docs: content/pytorch/concepts/tensor-operations/terms/tensor-split/tensor-split.md

---
Title: 'tensor-split'
Description: 'Takes a input for a tensor and splits it into multiple tensors.'

> Subjects:
>   - 'Data Science'

Reviewer: ```suggestion
Subjects:
  - 'AI'
  - 'Data Science'
  - 'Machine Learning'
```
Added some more subjects that are supported by `tensor-split` term in `tensor-operations.`

Tags:

>   - 'Data Structures'

Reviewer: ```suggestion
  - 'Data Structures'
  - 'Deep Learning'
  - 'PyTorch'
  - 'Tensor'
```
Added some more tags for better understanding and relevance.

CatalogContent:
  - 'intro-to-py-torch-and-neural-networks'
  - 'paths/data-science'
---

> In PyTorch, the **`.tensor_split()`** function splits a tensor into multiple sub-tensors. PyTorch will attempt to split the tensor evenly but when the tensor cannot be split evenly it will create as close to even sub-tensors as possible. The sub tensors can be also shaped with a dim (dimension) input.

Reviewer: ```suggestion
In PyTorch, the **`.tensor_split()`** function splits a tensor into multiple sub-tensors. If the tensor cannot be split evenly, PyTorch distributes the elements as evenly as possible. The shape of the sub-tensors can also be controlled using the `dim` (dimension) parameter.
```
Reframed the sentence for more consistency and grammatical correctness.

## Syntax

```pseudo
torch.tensor_split(input, indices_or_sections, dim=0)
```

- `input(Tensor)`: The tensor to be split.

> 

Reviewer: ```suggestion

```
Please remove this extra blank line over here.

> - `indices_or_sections(int)`: How many sub tensors to split into

Reviewer: ```suggestion
- `indices_or_sections(int)`: Number of sub-tensors to split the input tensor into, or the indices at which to split along the specified dimension.
```
This sentence seems grammatically incorrect or incomplete. So rephrase it for more clarity.

> 

Reviewer: ```suggestion

```
Please remove this extra blank line over here.

> - `dim(int, optional)`: Change the dimensions of the split tensors, Default=0

Reviewer: ```suggestion
- `dim(int, optional)`: The dimension along which to split the tensor. Default is `0`.
```

## Example

> The following example demonstrates the usage of the `.tensor_split()` function:

Reviewer: ```suggestion
The following example demonstrates the use of the `.tensor_split()` function:
```

```py
import torch

x = torch.arange(10) #create a one dimensional tensor
torch.tensor_split(x, 2) #split the tensor into 2 parts
```
```shell

> #the above code produces the tensor in 2 equal parts. 10 goes into 2 5 times so each subtensor has 5 elements

Reviewer: ```suggestion
#the above code produces the tensor in 2 equal parts. 10 goes into 2 5 times so each subtensor has 5 elements
```
Instead of this comment: "#the above code produces the tensor in 2 equal parts. 10 goes into 2 5 times so each sub-tensor has 5 elements," you can use the following sentence before the shell block: "The following code splits the tensor into 2 equal sub-tensors. Since 10 can be evenly divided by 2, each sub-tensor will contain 5 elements:"

(tensor([0, 1, 2, 3, 4]), tensor([5, 6, 7, 8, 9]))
```

> ```py
> import torch
> 
> x = torch.arange(13) #create a one dimensional tensor
> torch.tensor_split(x, 6) #split the tensor into 6 parts
> ```
> ```shell
> #the above code produces the tensor in 6 almost equal parts. 13 is not divisable by 6 so PyTorch does this as close to even as possible the first sub tensor having 3 elements and the rest only 2.
> (tensor([0, 1, 2]), tensor([3, 4]), tensor([5, 6]), tensor([7, 8]), tensor([ 9, 10]), tensor([11, 12]))
> ```

Reviewer: ```suggestion
```py
import torch

x = torch.arange(13) #create a one dimensional tensor
torch.tensor_split(x, 6) #split the tensor into 6 parts
```
```shell
#the above code produces the tensor in 6 almost equal parts. 13 is not divisable by 6 so PyTorch does this as close to even as possible the first sub tensor having 3 elements and the rest only 2.
(tensor([0, 1, 2]), tensor([3, 4]), tensor([5, 6]), tensor([7, 8]), tensor([ 9, 10]), tensor([11, 12]))
```
```
This section can be removed since an example has already been added earlier in this entry.
